### PR TITLE
Improve the migration guides

### PR DESCRIPTION
**Pull request by tyrannokapi**
This improves the migration guides, as requested in #144.

This is a WIP, so suggestions and feedback are of course welcome.

### CODE REVIEW

**Comment by boformer** [comment]
A cookie for you :cookie: 

Nice work!

File: source/server/getting-started/installation.rst [hunk 1]
```diff
@@ -14,24 +14,54 @@ Installing Sponge
...
+
+Adding Mods and Plugins
+=======================
+
+Sponge does not do much to your server without Forge mods or Sponge plugins. Players connecting to your server do not need to have any mods installed on their Minecraft clients if you plan to use Sponge plugins exclusively.
+
+Installing Forge Mods
+~~~~~~~~~~~~~~~~~~~~~
+
+There are several places around the internet to download Forge mods from. One of the most popular places is the Minecraft forums.
```
Comment (progwml6) [inline source/server/getting-started/installation.rst]: curseforge is another example

**Comment by BlackGoat** [comment]
@tyrannokapi Is this ready to lose the WIP and be merged?

**Comment by tyrannokapi** [comment]
@BlackGoat Yes. The SpongeVanilla migration guides still need a bit of work, but SpongeVanilla isn't available so I don't think those guides are a priority at the moment.

**Pull request merged**